environment/src/test/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionInitializationHandlerTest.java

```diff
@@ -137,6 +158,42 @@ void acceptTestEnvironmentShouldNotBeUpdatedWhenEncryptionKeyUrlIsEmpty() {
         verifyEnvCreationEvent();
     }
 
+    @Test
+    void acceptTestEnvironmentShouldBeUpdatedWhenEncryptionKeyUrlIsPresent() {
+        doAnswer(i -> null).when(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), any(Headers.class));
+        Environment environment = new Environment();
+        environment.setParameters(newAzureParameters());
+        when(environmentService.findEnvironmentById(ENVIRONMENT_ID)).thenReturn(Optional.of(environment));
+        CreatedDiskEncryptionSet createdDiskEncryptionSet = new CreatedDiskEncryptionSet.Builder()
+                .withDiskEncryptionSetId("dummmy-set-id")
+                .build();
+        when(environmentEncryptionService.createEncryptionResources(any(), any())).thenReturn(createdDiskEncryptionSet);
+
+        underTest.accept(environmentDtoEvent);
+
+        verify(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), headersArgumentCaptor.capture());
+        AzureParameters azureParameters = (AzureParameters) environment.getParameters();
+        assertEquals(azureParameters.getDiskEncryptionSetId(), "dummmy-set-id");
+        assertEquals(environment.getStatus(), EnvironmentStatus.ENVIRONMENT_ENCRYPTION_RESOURCES_INITIALIZED);
+        verifyEnvCreationEvent();
+    }
+
+    @Test
+    void acceptTestEnvironmentShouldNotBeUpdatedWhenCreatedDiskEncryptionSetIdIsNull() {
```

Comment:
As mentioned above, I do not feel it is a valid scenario to have `EnvironmentEncryptionService.createEncryptionResources()` return `null` (and with the current logic, it seems to be impossible to encounter such a situation).